Menu/Menu.js

```diff
@@ -1,10 +1,36 @@
 
+
+
+// Start Here: Create a reference to the ".menu" class
+ const menu = document.querySelector('.menu');
+// create a reference to the ".menu-button" class
+ const menuButton = document.querySelector('.menu-button');
+var a = 0;
+
 const toggleMenu = () => {
   // Toggle the "menu--open" class on your menu refence. 
+  a++;
+
+  if (a % 2 == 1)
+    menu.style.display = "flex";
+  else 
+    menu.style.display = "none";
 }
 
-// Start Here: Create a reference to the ".menu" class
-const menu;
-// create a reference to the ".menu-button" class
-const menuButton;
 // Using your menuButton reference, add a click handler that calls toggleMenu
+menuButton.addEventListener("mousedown", e => {
+  console.log(`You clicked on the menu button!`);
+  // menuButton.style.color = `blue`;
+  // menuButton.style.border = `5px solid black`;
+  toggleMenu();
+ 
+});
+
+// Using your menuButton reference, add a click handler that calls toggleMenu
+menuButton.addEventListener("mouseleave", e => {
+  console.log(`You clicked on the menu button!`);
+  // menuButton.style.color = `blue`;
+  // menuButton.style.border = `5px solid black`;
+  menu.style.display = "none";
```

Comment:
Your solution was interesting, but there's a problem. When your mouse leaves the hamburger menu, the menu closes. This would make it impossible to click on any of the menu items. 

But even if you managed to fix that, there's another problem. Your toggleMenu function uses an accumulator 'a' to determine whether the menu should be open or closed, based on whether the value stored at 'a' is even or odd. However, if you close the menu by mouseleave, the accumulator 'a' doesn't change. This makes it so you have to click on the hamburger twice to bring the menu back.